client/actions/show-heartbeat/index.js

```diff
@@ -128,11 +128,11 @@ export default class ShowHeartbeatAction extends Action {
     // A bit redundant but the action argument names shouldn't necessarily rely
     // on the argument names showHeartbeat takes.
     const heartbeatData = {
-      surveyId,
+      surveyId: `${surveyId}::${flow.id}`,
```

Comment:
Ah, the UUID we want to append isn't the flow ID. The flow ID is unique per-interaction with the prompt, so the same user interacting with different surveys will have a different flow ID. We want to use the per-user ID that doesn't change so that S&I can target and correlate answers from the same users over time.

We want to use the `userId` made available to filter expressions (see the [filter expression docs](http://normandy.readthedocs.io/en/latest/user/filter_expressions.html#normandy.userId)). The `getUserId` function in `self_repair_runner.js` can fetch it. I suspect we probably want to include it in the client info returned from `driver.client()` so that actions can use it.